### PR TITLE
pitchfork: Add pitchfork plugin

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/coala/corobo/issues/36

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: plugins/pitchfork.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        """
+        To pitchfork user down to ...
+        """
+        match = re.match(r'@?([\w-]+)(?:\s+(?:down\s+)?to\s+(.+))?$',
+                         arg)
+        if match:
+            user = match.group(1)
+            place = match.group(2) if match.group(2) else 'offtopic'
+            return (
+                string.Template("""
```
Comment (NiklasMM) [inline plugins/pitchfork.py]: I find unindented multi line strings disrupt the flow when reading a function/file. You could use [dedent](https://docs.python.org/3.6/library/textwrap.html#textwrap.dedent) to have the literal be properly indented and still not have the indentation in the actual string.

However then you'll probably run into the line length limit.... Hmm.... maybe make the pitchfork a bit shorter? I did notice that on mobile I just saw the stick: 
screenshot_20170726-083940

**Comment by meetmangukiya** [comment]
@NiklasMM PTAL?

File: plugins/pitchfork.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        To pitchfork user down to ...
+        """
+        match = re.match(r'@?([\w-]+)(?:\s+(?:down\s+)?to\s+(.+))?$',
+                         arg)
+        if match:
+            user = match.group(1)
+            place = match.group(2) if match.group(2) else 'offtopic'
+            return textwrap.dedent((
+                string.Template("""
+                    @$user, you are being pitchforked down to $place
```
Comment (SanketDG) [inline plugins/pitchfork.py]: if its oftopic, we could make it a markdown link? that links to the offtopic channel

**Comment by meetmangukiya** [comment]
@SanketDG PTAL

File: plugins/pitchfork.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (SanketDG) [inline plugins/pitchfork.py]: any reason to use `string.Template` over https://pyformat.info?

Comment (meetmangukiya) [reply plugins/pitchfork.py]: yes, because the string contains `{` and `}` and there's no way to escape those(can't believe it)

Comment (SanketDG) [reply plugins/pitchfork.py]: um you can put `{{` `}}` to escape them, but this is fine, I guess

**Comment by meetmangukiya** [comment]
@SanketDG can you approve if there's nothing else left?

**Comment by SanketDG** [comment]
ack f7363b0

**Comment by meetmangukiya** [comment]
reack 28f72fb

**Comment by meetmangukiya** [comment]
@rultor merge

**Pull request merged**